python/python/bystro/api/proteomics.py

```diff
@@ -0,0 +1,185 @@
+            experiment_names = set()
+            for line in file:
+                row = line.strip().split("\t")
+                experiment_names.add(row[experiment_name_index])
+
+        if len(experiment_names) == 1:
+            experiment_name = experiment_name or experiment_names.pop()
+        elif not experiment_name:
+            raise ValueError(
+                "Either no experiment name or multiple experiment names "
+                "were found in the experiment annotation file. "
+                "Please provide an experiment name."
+            )
+
+        state, auth_header = authenticate()
+        experiment_file = _package_filename(experiment_annotation_file)
+
+        experiment_response = requests.post(
+            state.url + EXPERIMENT_ENDPOINT,
+            headers=auth_header,
+            files={"file": experiment_file[1]},
+            data={"experimentName": experiment_name},
+        )
+        if experiment_response.status_code != HTTP_STATUS_OK:
+            raise Exception(
+                "Experiment annotation upload failed with status "
+                f"code {experiment_response.status_code}: {experiment_response.text}"
+            )
+
+        if print_result:
+            print(
+                "\nExperiment Annotation Upload Response:",
+                json.dumps(experiment_response.json(), indent=4),
+            )
+
+    state, auth_header = authenticate()
+    url = state.url + UPLOAD_PROTEIN_ENDPOINT
+
+    if annotation_job_id:
+        annotation_url = state.url + GET_ANNOTATION.replace(":id", str(annotation_job_id))
+        annotation_response = requests.get(annotation_url, headers=auth_header)
+
+        if annotation_response.status_code != HTTP_STATUS_OK:
+            raise RuntimeError(
+                f"The annotation with ID {annotation_job_id} "
+                "does not exist or you do not have permissions to access this annotation."
+            )
+
+    job_payload = {
+        "proteomics_dataset_type": proteomics_dataset_type.value,
+        "assembly": "NA",
+        "proteomics": {"annotationLink": annotation_job_id},
+        "experimentName": experiment_name,
+    }
+
+    protein_files = [_package_filename(protein_abundance_file)]
+
+    response = requests.post(
+        url, headers=auth_header, files=protein_files, data={"job": json.dumps(job_payload)}
+    )
+
+    if response.status_code != HTTP_STATUS_OK:
+        raise RuntimeError(f"Upload failed with status code {response.status_code}: {response.text}")
+
+    proteomics_response_data = response.json()
+    if print_result:
+        print("\nProteomics Upload Response:", json.dumps(proteomics_response_data, indent=4))
+
+    proteomics_job_id = proteomics_response_data.get("_id")
+    final_response = {"proteomicsID": proteomics_job_id}
+
+    if annotation_job_id:
+        update_annotation_payload = {"proteomicsID": proteomics_job_id}
```

Comment:
This doesn't seem to work. I would actually prefer to remove it for now, because I don't have a strong use case for the link in the reverse direction

Reply:
Let's do this:

```py
    if annotation_job_id:
        final_response["annotationID"] = annotation_job_id
        if print_result:
            annotation_name = annotation_response.json().get("name", "Unknown Annotation")
            print(
                f'\nAnnotation "{annotation_name}" (id: {annotation_job_id}) '
                f'was linked to this proteomics submission (id: "{proteomics_job_id}").\n'
            )
```

This will skip patching the annotation job, and just print out what was linked after fetching the name